#Variant_analysis

#Wczytywanie_pliku
library("plyr")
library("tidyr")
library("ff")
library("ggplot2")
library(reshape2)
library(scales)
library(wesanderson)
library(biglm)
library(ffbase)
#ktory_plik_do_czego
#data_1.txt - orginalny z pelnymi nazwami wariantow
#gnomad_all_chromo_2 -  tylko z 1 rodzajem wariantu
#gnomad_new_cat_2 - zredukowane warianty

#wszystkie_warianty
gnomad_all_chromo <- read.csv.ffdf(file="gnomad_all_chromo_2.txt",sep='\t', VERBOSE=TRUE, first.rows=10000,next.rows=50000, colClasses=NA)

#warianty_bia³kowe
gnomad_all_chromo <- read.csv.ffdf(file="gnomad_new_cat_2.txt",sep='\t', VERBOSE=TRUE, first.rows=10000,next.rows=50000, colClasses=NA)



#TABELE
#tabela z AN
table <- read.table('table_AN_2.csv', sep=',', head=TRUE) 
table2 <- melt(table, id.vars = 'Chromosom')

#tabela z AF_AN - rzadkie warianty < 0.01
table_AN_AF <- read.table('table_AF_AN_2.csv', sep='\t', head=FALSE) 
names(table_AN_AF) = c("Chromosom","Wariant","AF","AN")

#Tworzenie tabeli z iloœcia wariantów
variant_table <- table(gnomad_all_chromo[,1],gnomad_all_chromo[,8])
write.table(variant_table, file = "table_AN_full.csv", sep = ",")

table <- read.table('table_AN_full.csv', sep=',', head=TRUE)
table2 <- melt(table, id.vars = 'Chromosom')
names(table2) = c("Chromosom","Wariant","AN")

table <- read.table('table_AN_full_cutted_2.csv', sep=',', head=TRUE)
table2 <- melt(table, id.vars = 'Chromosom')
names(table2) = c("Chromosom","Wariant","AN")

#
#table <- read.table('table_AN_2.csv', sep=',', head=TRUE)
#table2 <- melt(table, id.vars = 'Chromosom')
#names(table2) = c("Chromosom","Wariant","AN")

#przetwarzanie danych dla AF<0.01 
table3 = merge(mean_freq, table2, by=c("Chromosom","Wariant")) #najpierw wygenerowac table2(AN) i mean_freq(AF)
write.table(table3, file = "table_AF_AN.csv", sep = "\t",row.names = F,col.names = F) 



#Formatowanie_kolumn
names(gnomad_all_chromo)[1] <- "Chromosom"
names(gnomad_all_chromo)[6] <- "AN"
names(gnomad_all_chromo)[7] <- "AF"
names(gnomad_all_chromo)[8] <- "Variant"

#Data_frame
mean_freq <- aggregate(gnomad_all_chromo[,7],list(gnomad_all_chromo[,1],gnomad_all_chromo[,8]),mean)
names(mean_freq)[1]<-"Chromosom"
names(mean_freq)[2]<-"Wariant"
names(mean_freq)[3]<-"Frekwencja"

mean_allel_number <- aggregate(gnomad_all_chromo[,6],list(gnomad_all_chromo[,1],gnomad_all_chromo[,8]),mean)
names(mean_allel_number)[1]<-"Chromosom"
names(mean_allel_number)[2]<-"Wariant"
names(mean_allel_number)[3]<-"value"

table3 = merge(mean_freq, table2, by=c("Chromosom","Wariant"))

allel_number <- aggregate(gnomad_all_chromo[,6],list(gnomad_all_chromo[,8]),sum)
names(allel_number)[1]<-"Wariant"
names(allel_number)[2]<-"Ilosc"


#srednia AF z ca³ego genomu

mean_freq_whole_genome <- aggregate(gnomad_all_chromo[,7],list(gnomad_all_chromo[,1],gnomad_all_chromo[,8]),sum)
names(mean_freq_whole_genome)[1]<-"Chromosom"
names(mean_freq_whole_genome)[2]<-"Wariant"
names(mean_freq_whole_genome)[3]<-"Frekwencja"
mean_freq_whole_genome$Frekwencja = mean_freq_whole_genome$Frekwencja/dim(gnomad_all_chromo)[1]





#WYKRESY

#BAR_PLOTS
#AN
ggplot(table2,aes(x = Chromosom, y = AN, fill = Wariant)) + geom_bar(stat = "identity", colour = "black") + xlab("\nChromosom") + ylab("Liczba_wariantów\n") + guides(fill=guide_legend(title="Rodzaj wariantu")) + scale_x_continuous(breaks=1:22) 

ggplot(table2,aes(x = Chromosom, y = AN, fill = Wariant)) + geom_bar(position = "fill",stat = "identity",colour = "black") + scale_y_continuous(labels = percent_format()) + guides(fill=guide_legend(title="Rodzaj wariantu")) + xlab("\nChromosom") + ylab("Liczba_wariantów(w %)\n") + scale_x_continuous(breaks=1:22) 


#Stacked_bar_AF
ggplot(mean_freq,aes(x = Chromosom, y = Frekwencja, fill = Wariant)) + geom_bar(position = "fill",stat = "identity",colour = "black")  + guides(fill=guide_legend(title="Rodzaj wariantu")) + xlab("\nChromosom") + ylab("Frekwencja\n") + scale_x_continuous(breaks=1:22) 

#rare_wariats_<_0.01 
ggplot(table_AN_AF,aes(x = Chromosom, y = AN, fill = Wariant)) + geom_bar(stat = "identity", colour = "black") + xlab("\nChromosom") + ylab("Liczba_wariantów\n") + guides(fill=guide_legend(title="Rodzaj wariantu")) + scale_x_continuous(breaks=1:22) 

#Stacked_bar_AF_whole_genome_mean
ggplot(mean_freq_whole_genome,aes(x = Chromosom, y = Frekwencja, fill = Wariant)) + geom_bar(position = "fill",stat = "identity",colour = "black")  + guides(fill=guide_legend(title="Rodzaj wariantu")) + xlab("\nChromosom") + ylab("Frekwencja\n") + scale_x_continuous(breaks=1:22) 

ggplot(data=allel_number, aes(x=Wariant, y=Ilosc)) + geom_bar(stat="identity",fill = "#265fe2") + geom_text(aes(label = Ilosc,vjust=-1)) + theme(axis.text.x = element_text(angle = 45, hjust = 1))

#brzydkie
ggplot(mean_freq,aes(x = Chromosom, y = Frekwencja)) + geom_bar(stat = "identity", fill = "#265fe2") + xlab("\nChromosom") + ylab("Frekwencja(w %)\n") + scale_y_continuous(labels = percent_format())
#nie_³adne
ggplot(mean_allel_number,aes(x = Chromosom, y = value)) + geom_bar(stat = "identity", fill = "#265fe2") + xlab("\nChromosom") + ylab("Œrednia iloœæ przebadanych alleli\n")

ggplot(data=mean_allel_number, aes(x=Chromosom, y=value, fill=Wariant)) +
  geom_bar(stat = "identity",colour = "black") + scale_x_continuous(breaks=1:22) 


ggplot(data=mean_freq, aes(x=Chromosom, y=Frekwencja, fill=Wariant)) +
  geom_bar(stat = "identity",colour = "black") + scale_x_continuous(breaks=1:22) 
  

names(gnomad_all_chromo[1])
names(gnomad_all_chromo) <- c('Chrom','Pos','Ref','Alt','AC','AN','AF','Variant','Impact','Gene_ID','Ens_Gene_ID','Unknown','Ens_trans','SNV')

#linear_model
lm = bigglm(AN~Chrom+Variant,data=gnomad_all_chromo,chunksize=50000)

#aov = aov(AN~Chromosom+Wariant,data=gnomad_all_chromo)


